streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -4017,6 +4017,60 @@ class ZStream[-R, +E, +A](val channel: ZChannel[R, Any, Any, Any, E, Chunk[A], A
   def whenZIO[R1 <: R, E1 >: E](b: ZIO[R1, E1, Boolean])(implicit trace: ZTraceElement): ZStream[R1, E1, A] =
     ZStream.whenZIO(b)(self)
 
+  /**
+   * Emits a sliding window of n elements.
+   */
+  def windowed(chunkSize: Int)(implicit trace: ZTraceElement): ZStream[R, E, Chunk[A]] =
```

Comment:
I think I might call this `grouped`. I believe that is the name for the Scala collection operation this corresponds to and I think normally we have found it has worked well to give stream versions of collection operators the same name. There is also potentially a more general version of this `sliding` that allows the step size to be different than the window size.

A further question for at least this version is how it related to chunking. Can this just be rewritten as `chunkN(chunkSize).chunks`. That is at least in its simple form where the step size and window size are the same I think it just amounts to rechunking the stream into the window size and then exposing those chunks.

Reply:
Ah I see this uses a step size of one instead of the window size. Yes I think `sliding` is the right name for this though would be good to be able to control the step size in addition to the window size.

Reply:
Renamed to `sliding`. As suggested, now it includes a stepSize parameter.